driver-core/src/main/com/mongodb/connection/Server.java

```diff
@@ -54,5 +54,5 @@
      *
      * @param callback the callback to execute when the connection is available or an error occurs
      */
-    void getConnectionAsync(SingleResultCallback<Connection> callback);
+    void getConnectionAsync(SingleResultCallback<AsyncConnection> callback);
```

Comment:
getAsyncConnectionAsync is just awful though.  And all of our async methods end in Async.  So the way it is seems the least of all evils.